test/e2e/flex_to_dedicated_test.go

```diff
@@ -0,0 +1,287 @@
+// Copyright 2025 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e_test
+
+import (
+	"context"
+	"os/exec"
+	"syscall"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/api"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/api/v1"
+	akov2common "github.com/mongodb/mongodb-atlas-kubernetes/v2/api/v1/common"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/e2e/k8s"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/e2e/utils"
+)
+
+var _ = Describe("Flex to Dedicated Upgrade", Ordered, Label("flex-to-dedicated"), func() {
+	var kubeClient client.Client
+	var testNamespace *corev1.Namespace
+	var resourcePrefix string
+	var akoCmd *exec.Cmd
+
+	_ = BeforeAll(func(ctx context.Context) {
+		cmd, err := k8s.RunManagerBinary(false)
+		Expect(err).To(BeNil())
+		akoCmd = cmd
+
+		client, err := k8s.CreateNewClient()
```

Comment:
This library here has a lot of luggage:
- This call us OK, but it is not experimental schemas aware, while the new kube library is.
- There are a lot of adhoc functions there that hide many Kubernetes interactions: `GetPodStatus`, `GetDeploymentObservedGeneration` etc

Even if we do agree on not using those calls, they will still be there and complicate clean up later, or even make it not happen at all.

Reply:
Again, I copied only what was of my interest to run the test I introduced. Bring full functionality from what you did was out of my scope.